PhysicsTools/PatAlgos/src/SoftMuonMvaEstimator.cc

```diff
@@ -0,0 +1,153 @@
+
+	tmvaReader_.AddSpectator("pID",			       	&pID_);
+	tmvaReader_.AddSpectator("pt",				&pt_);
+	tmvaReader_.AddSpectator("eta",				&eta_);
+	tmvaReader_.AddSpectator("MomID",			&momID_);
+
+	tmvaReader_.BookMVA("BDT", weightsfile);
+	initialized_ = true;	
+}
+
+void SoftMuonMvaEstimator::computeMva(const pat::Muon& muon)
+{
+	if (not initialized_)
+		throw cms::Exception("FatalError") << "SoftMuonMVA is not initialized";
+
+	reco::TrackRef gTrack = muon.globalTrack();
+	reco::TrackRef iTrack = muon.innerTrack();
+	reco::TrackRef oTrack = muon.outerTrack();
+
+	if(!(muon.innerTrack().isNonnull() and 
+	     muon.outerTrack().isNonnull() and 
+	     muon.globalTrack().isNonnull()))
+	{
+	  mva_ = -1; 
+	  return;
+	}
+
+	//VARIABLE EXTRACTION
+	pt_ = muon.pt();
+	eta_ = muon.eta();
+	dummy_ = -1;
+	momID_ = dummy_;
+	pID_ = dummy_;
+
+
+	chi2LocalMomentum_ = muon.combinedQuality().chi2LocalMomentum;
+	chi2LocalPosition_ = muon.combinedQuality().chi2LocalPosition;
+	glbTrackProbability_ =  muon.combinedQuality().glbTrackProbability;
+	trkRelChi2_ =  muon.combinedQuality().trkRelChi2;
+
+	trkKink_ = muon.combinedQuality().trkKink;
+	log2PlusGlbKink_ =  TMath::Log(2+muon.combinedQuality().glbKink);
+	segmentCompatibility_ = muon.segmentCompatibility();
+
+	timeAtIpInOutErr_ = muon.time().timeAtIpInOutErr;
+
+	//TRACK RELATED VARIABLES
+	
+	iValidFraction_ =  iTrack->validFraction();
+	innerChi2_ =  iTrack->normalizedChi2();
+	layersWithMeasurement_ = iTrack->hitPattern().trackerLayersWithMeasurement();
+
+	outerChi2_ = oTrack->normalizedChi2();
+
+	qProd_ =  iTrack->charge()*oTrack->charge();
+
+	const reco::HitPattern &gMpattern = gTrack->hitPattern();
+
+	std::vector<int> fvDThits = {0,0,0,0};
+	std::vector<int> fvRPChits = {0,0,0,0};
+	std::vector<int> fvCSChits = {0,0,0,0};
+
+	vMuonHitComb_ = 0;
+
+	for (int i=0;i<gMpattern.numberOfAllHits(reco::HitPattern::TRACK_HITS);i++){
+
+	  uint32_t hit = gMpattern.getHitPattern(reco::HitPattern::TRACK_HITS, i);
+	  if ( !gMpattern.validHitFilter(hit) ) continue;
+
+	  if (gMpattern.getMuonStation(hit) == 1){
+	    if (gMpattern.muonDTHitFilter(hit)) fvDThits[0]++;
+	    if (gMpattern.muonRPCHitFilter(hit)) fvRPChits[0]++;
+	    if (gMpattern.muonCSCHitFilter(hit)) fvCSChits[0]++;
+	  }
+  	
+	  if (gMpattern.getMuonStation(hit) == 2){
+	    if (gMpattern.muonDTHitFilter(hit)) fvDThits[1]++;
+	    if (gMpattern.muonRPCHitFilter(hit)) fvRPChits[1]++;
+	    if (gMpattern.muonCSCHitFilter(hit)) fvCSChits[1]++;
+	  }
+
+	  if (gMpattern.getMuonStation(hit) == 3){
+	    if (gMpattern.muonDTHitFilter(hit)) fvDThits[2]++;
+	    if (gMpattern.muonRPCHitFilter(hit)) fvRPChits[2]++;
+	    if (gMpattern.muonCSCHitFilter(hit)) fvCSChits[2]++;
+	  }
+
+	  if (gMpattern.getMuonStation(hit) == 4){
+	    if (gMpattern.muonDTHitFilter(hit)) fvDThits[3]++;
+	    if (gMpattern.muonRPCHitFilter(hit)) fvRPChits[3]++;
+	    if (gMpattern.muonCSCHitFilter(hit)) fvCSChits[3]++;
+	  }
```

Comment:
can the above be simply 
```c++
int muStation0 = gMpattern.getMuonStation(hit) - 1;
if (muStation0 >=0 && muStation0 < 4){
if (gMpattern.muonDTHitFilter(hit)) fvDThits[muStation0]++;
if (gMpattern.muonRPCHitFilter(hit)) fvRPChits[muStation0]++;
if (gMpattern.muonCSCHitFilter(hit)) fvCSChits[muStation0]++;
}

```

?

Reply:
Yes. Thank you for the suggestion, I will change it in the next commit.